iothub/device/src/Authentication/HsmAuthentication/HttpHsmSignatureProvider.cs

```diff
@@ -90,15 +84,12 @@ public async Task<string> SignAsync(string moduleId, string generationId, string
             string generationId,
             SignRequest signRequest)
         {
-            var transientRetryPolicy = new RetryPolicy(s_transientErrorDetectionStrategy, s_transientRetryStrategy);
-            return await transientRetryPolicy
-                .RunWithRetryAsync(() => hsmHttpClient.SignAsync(_apiVersion, moduleId, generationId, signRequest))
+            var retryHandler = new RetryHandler(s_retryPolicy);
+            return await retryHandler
+                .RunWithRetryAsync(
+                    () => hsmHttpClient.SignAsync(_apiVersion, moduleId, generationId, signRequest),
+                    (Exception ex) => ex is SwaggerException se && se.StatusCode >= 500)
                 .ConfigureAwait(false);
         }
-
-        private class ErrorDetectionStrategy : ITransientErrorDetectionStrategy
```

Comment:
Made a flavor of the RetryHandler API to pass code-path-specific overrides for what is transient.